pyomo/contrib/mindtpy/util.py

```diff
@@ -631,27 +648,21 @@ def copy_var_list_values_from_solution_pool(from_list, to_list, config, solver_m
                 solver_model.setParam(
                     gurobipy.GRB.Param.SolutionNumber, solution_name)
                 var_val = var_map[v_from].Xn
-            else:
-                config.logger(
-                    config.mip_solver + 'is not supported for solution pool in Mindtpy yet.')
-                raise NotImplementedError()
             v_to.set_value(var_val)
-            if skip_stale:
-                v_to.stale = False
         except ValueError as err:
             err_msg = getattr(err, 'message', str(err))
             rounded_val = int(round(var_val))
             # Check to see if this is just a tolerance issue
             if ignore_integrality \
-                    and v_to.is_integer():  # not v_to.is_continuous()
+                    and v_to.is_integer():
                 v_to.value = var_val
-            elif v_to.is_integer() and (abs(var_val - rounded_val) <= config.integer_tolerance):  # not v_to.is_continuous()
+            elif v_to.is_integer() and (abs(var_val - rounded_val) <= config.integer_tolerance):
                 v_to.set_value(rounded_val)
             elif 'is not in domain NonNegativeReals' in err_msg and (
                     abs(var_val) <= config.zero_tolerance):
                 v_to.set_value(0)
             else:
-                raise
+                raise ValueError
```

Comment:
Generally, this should raise an instance of a ValueError and not the ValueError class.  Further, as you are already in an exception that is raising a ValueError, I would recommend logging a message and re-raising the original exception:
```python
else:
    config.logger.error('Unknown validation domain error setting variable %s', (v_to.name,))
    raise
```